src/core/zip/MCZipMac.mm

```diff
@@ -0,0 +1,31 @@
+//
+//  MCZipMac.mm
+//  mailcore2
+//
+//  Created by juliangsp on 7/7/15.
+//  Copyright (c) 2015 MailCore. All rights reserved.
+//
+
+#import "MCZipMac.h"
+#import "NSObject+MCO.h"
+#import <Foundation/Foundation.h>
+
+using namespace mailcore;
+
+String * mailcore::TemporaryDirectoryForZip()
+{
+    NSError * error;
+    NSString * newDirString;
+    NSURL * directoryURL;
+    
+    error = nil;
+    newDirString = [NSTemporaryDirectory() stringByAppendingPathComponent:[[NSProcessInfo processInfo] globallyUniqueString]];
+    directoryURL = [NSURL fileURLWithPath:newDirString isDirectory:YES];
+    
+    [[NSFileManager defaultManager] createDirectoryAtURL:directoryURL withIntermediateDirectories:YES attributes:nil error:&error];
```

Comment:
After a second thought, could you append a [[NSUUID UUID] UUIDString] to the path so we can ensure it's even a unique path for this process? We need to do it because we are removing this folder when the zip file is created.